scripts/ado-terraform-nagger.py

```diff
@@ -272,155 +399,333 @@ def extract_version(text, regex):
 
     try:
-        # Try to run `version --json` which is present in tf versions >= 0.13.0
-        result = json.loads(run_command(command))
-        terraform_version = result["terraform_version"]
-
-        # Use terraform's `terraform_outdated` JSON object to notify if there
-        # is a new terraform version available.
-        if "terraform_outdated" in result and result["terraform_outdated"]:
-            log_message(
-                None,
-                None,
-                "warning",
-                f"Detected outdated terraform version: {terraform_version}",
-            )
-
-        # Handle terraform versions
-        terraform_version_checker(terraform_version)
-
-        # Handle providers
-        terraform_providers = result["provider_selections"]
-
-        for provider in terraform_providers:
-            if provider not in config["providers"]:
-                log_message(
+        # Retrieve the environment variables
+        system_default_working_directory = os.getenv('SYSTEM_DEFAULT_WORKING_DIRECTORY')
+        build_repo_suffix = os.getenv('BUILD_REPO_SUFFIX')
+
+        # Construct the working directory path
+        base_directory = os.getenv('BASE_DIRECTORY')
+        working_directory, components_list = create_working_dir_list(base_directory, system_default_working_directory, build_repo_suffix)
+        
+        output_warning = {
+            'terraform_version': {
+                'components': [],
+                'error_message': ''
+            },
+            'terraform_provider': {
+                'provider': {},
+                'error_message': ''
+            }
+        }
+
+        # for loop over dir componenets
+        for component in components_list:
+            print(f'COMPONENT: {component}')
+            full_path = f'{working_directory}{component}'
+            print(f'FULL PATH: {full_path}')
+
+            # Load deprecation map
+            config = load_file(args.filepath)
+
+            # Try to run `tfswitch' and 'terraform version --json` which is present in tf versions >= 0.13.0
+            command = ["tfswitch", "-b", terraform_binary_path]
+            run_command(command, full_path)
+
+            # try and do the terraform init 
+            command = ["terraform", "init", "-backend=false"]
+            output = run_command(command, full_path)
+            # check if tf has successfully init 
+            if not 'Terraform has been successfully initialized!' in output:
+                global errors_detected
+                relative_test_path = os.path.relpath(full_path, '../../../../../azp/_work/1/s')
+                logger.error(f'##vso[task.logissue type=error;]Repo structure invalid please see docs for further information: {relative_test_path}')
+                message = (f'Terraform initialized in an empty directory, Repo structure invalid please see docs for further information:\n{relative_test_path}')
```

Comment:
```suggestion
                message = (f'Repo structure invalid for use with terraform nagger. Please see docs for further information:\n{relative_test_path}')
```

Think this could be a bit more generic now we're testing the positive case